src/azure_cosmosdb/cosmosdb_utils.bal

```diff
@@ -0,0 +1,296 @@
+    }
+    if requestOptions.ifMatch is string {
+        request.setHeader(IF_MATCH_HEADER, requestOptions.ifMatch.toString());
+    }
+    if requestOptions.enableCrossPartition == true {
+        request.setHeader(IS_ENABLE_CROSS_PARTITION_HEADER, requestOptions.enableCrossPartition.toString());
+    }
+    return request;
+}
+
+isolated function setExpiryHeader(http:Request request, int validationPeriod) returns http:Request|error {
+    if validationPeriod >= 3600 && validationPeriod <= 18000 {
+        request.setHeader(EXPIRY_HEADER, validationPeriod.toString());
+        return request;
+    }else {
+        return prepareError("Resource token validity period must be between 3600 and 18000");
+    }
+}
+
+isolated function getTime() returns string?|error {
+    time:Time currentTime = time:currentTime();
+    var timeInTimeZone = check time:toTimeZone(currentTime, GMT_ZONE);
+    string|error timeString = time:format(timeInTimeZone, "EEE, dd MMM yyyy HH:mm:ss z");
+    if timeString is string {
+        return timeString;
+    } else {
+        return prepareError("Time is not correct");
+    }
+}
+
+isolated function generateMasterTokenSignature(string verb, string resourceType, string resourceId, string keyToken, string tokenType, 
+string tokenVersion, string date) returns string?|error {    
+    string authorization;
+    string payload = verb.toLowerAscii()+"\n" + resourceType.toLowerAscii() + "\n" + resourceId + "\n"
+    + date.toLowerAscii() +"\n" + "" + "\n";
+    var decoded = array:fromBase64(keyToken);
+    if decoded is byte[] {
+        byte[] digest = crypto:hmacSha256(payload.toBytes(),decoded);
+        string signature = array:toBase64(digest);
+        authorization = 
+        check encoding:encodeUriComponent(string `type=${tokenType}&ver=${tokenVersion}&sig=${signature}`, "UTF-8");   
+        return authorization;
+    } else {     
+        return prepareError("Base64 Decoding error");
+    }
+}
+
+isolated function mapResponseToTuple(http:Response|http:ClientError httpResponse) returns @tainted [json, Headers]|error {
+    var responseBody = check mapResponseToJson(httpResponse);
+    var responseHeaders = check mapResponseHeadersToObject(httpResponse);
+    return [responseBody,responseHeaders];
+}
+
+isolated function mapResponseToJson(http:Response|http:ClientError httpResponse) returns @tainted json|error { 
+    if (httpResponse is http:Response) {
+        var jsonResponse = httpResponse.getJsonPayload();
+        if (jsonResponse is json) {
+            if (httpResponse.statusCode != http:STATUS_OK && httpResponse.statusCode != http:STATUS_CREATED) {
+                return createResponseFailMessage(httpResponse,jsonResponse);
+            }
+            return jsonResponse;
+        } else {
+            return prepareError("Error occurred while accessing the JSON payload of the response");
+        }
+    } else {
+        return prepareError("Error occurred while invoking the REST API");
```

Comment:
Constant string can be used to address error messages as same message is used many places.